src/problem.cpp

```diff
@@ -0,0 +1,375 @@
+                            find1_solutions.data(),
+                            workspace.get(),
+                            workspace_size,
+                            options.exhaustive_search);
+        break;
+    }
+    case miopenProblemDirectionBackward: {
+        const auto method = conv_desc.mode == miopenTranspose
+                                ? &ConvolutionDescriptor::FindConvFwdAlgorithm
+                                : &ConvolutionDescriptor::FindConvBwdDataAlgorithm;
+
+        (conv_desc.*method)(handle,
+                            y_desc,
+                            y.get(),
+                            w_desc,
+                            w.get(),
+                            x_desc,
+                            x.get(),
+                            max_solutions,
+                            &found,
+                            find1_solutions.data(),
+                            workspace.get(),
+                            workspace_size,
+                            options.exhaustive_search);
+        break;
+    }
+    case miopenProblemDirectionBackwardWeights: {
+        decltype(auto) x_desc_ = conv_desc.mode == miopenTranspose ? y_desc : x_desc;
+        decltype(auto) x_      = conv_desc.mode == miopenTranspose ? y : x;
+        decltype(auto) y_desc_ = conv_desc.mode == miopenTranspose ? x_desc : y_desc;
+        decltype(auto) y_      = conv_desc.mode == miopenTranspose ? x : y;
+
+        conv_desc.FindConvBwdWeightsAlgorithm(handle,
+                                              y_desc_,
+                                              y_.get(),
+                                              x_desc_,
+                                              x_.get(),
+                                              w_desc,
+                                              w.get(),
+                                              max_solutions,
+                                              &found,
+                                              find1_solutions.data(),
+                                              workspace.get(),
+                                              workspace_size,
+                                              options.exhaustive_search);
+        break;
+    }
+    }
+
+    ret.reserve(found);
+
+    const auto conv_dir = static_cast<conv::Direction>(direction);
+    const auto netcfg   = AsConvolution().BuildConfKey();
+
+    for(auto i = 0; i < found; ++i)
+    {
+        const auto algo = ConvolutionAlgoToDirectionalString(
+            static_cast<miopenConvAlgorithm_t>(find1_solutions[i].fwd_algo), conv_dir);
+
+        auto solution = Solution{};
+        solution.SetTime(find1_solutions[i].time);
+        solution.SetWorkspaceSize(find1_solutions[i].memory);
+        solution.SetSolver(handle.GetFound1_0SolverId(netcfg, AlgorithmName{algo}).value());
+        solution.SetProblem(*this);
+
+        MIOPEN_LOG_I("Found solvution: " << solution.GetSolver().ToString() << " , "
```

Comment:
Typo

Reply:
@DrizztDoUrden  please fix in later PRs. Thanks!